sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicOperators.scala

```diff
@@ -513,3 +513,16 @@ case class MapGroups[K, T, U](
   override def missingInput: AttributeSet = AttributeSet.empty
 }
 
+case class CoGroup2[K, T, U, V](
```

Comment:
Also I'd probably just call it cogroup as we'll probably use a single variadic operator if we decide to do more than 2.